File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/utils/ResourceUtils.java
package com.firefly.example.reactive.coffee.store.utils;

import com.firefly.annotation.Component;
import com.firefly.utils.io.ClassRelativeResourceLoader;
import com.firefly.utils.io.Resource;

/**
 * @author Pengtao Qiu
 */
@Component
public class ResourceUtils {

    private final ClassRelativeResourceLoader resourceLoader = new ClassRelativeResourceLoader(getClass());

    public Resource resource(String path) {
        return resourceLoader.getResource(path);
    }
}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/utils/ResourceUtils.kt
package com.firefly.example.kotlin.coffee.store.utils

import com.firefly.annotation.Component
import com.firefly.utils.io.ClassRelativeResourceLoader
import com.firefly.utils.io.Resource

/**
 * @author Pengtao Qiu
 */
@Component
class ResourceUtils {

    private val resourceLoader = ClassRelativeResourceLoader(javaClass)

    fun resource(path: String): Resource {
        return resourceLoader.getResource(path)
    }
}